src/ua.c

```diff
@@ -31,6 +31,8 @@ struct ua {
 	int af_media;                /**< Preferred Address Family for media */
 	enum presence_status my_status; /**< Presence Status                 */
 	bool catchall;               /**< Catch all inbound requests         */
+	struct list *hdr_filter;     /**< Filter for incoming headers        */
```

Comment:
I think the filter can be converted to a compound list, that needs no allocation.

I have made an example of it here:

https://gist.github.com/alfredh/2577a5dadbde35a9ba47d96696ddb31d